tests/ui/feature-gates/feature-gate-never_patterns.rs

```diff
@@ -12,16 +12,42 @@ fn main() {
     unsafe {
         let ptr: *const Void = NonNull::dangling().as_ptr();
         match *ptr {
-            ! //~ ERROR `!` patterns are experimental
+            !
+            //~^ ERROR `!` patterns are experimental
+            //~| ERROR `!` patterns are experimental
```

Comment:
Why is this reported twice?
If `!` patterns are gated in the parser, then they don't need to be gated for the second time later on.

Reply:
It's gated the first time for the `!` syntax and the second time for the match arm without body. Because of how my check is done in `check_crate`, it emits the same emssage for both

Reply:
Fixed